front-end-babylonjs/src/game/levels/generators/ScamsGenerator.js

```diff
@@ -0,0 +1,461 @@
+                } 
+            }
+        }, 4000);
+    }
+
+    /**
+     * Function to create the scam object.
+     * @param {string} type - Flag to decide the behaviour of the scam.
+     */
+    createScams(type) {
+
+        // To position scam objects on different lanes randomly Default to Middle Lane
+        let randomPositionChooser = Math.floor((Math.random() * 100)); // 0 to 100 random number
+        let positionX = 0;
+        if (randomPositionChooser >= 0 && randomPositionChooser < 30) {
+            positionX = GAME.isMobile() ? -1 : -2.5; // Positining on the left
+        }
+
+        if (randomPositionChooser >= 30) {
+            positionX = 0;
+        }
+        if (randomPositionChooser >= 60) {
+            positionX = GAME.isMobile() ? 1 : 2.5; // Positioning on the right
+        }
+        let scamDiameter = GAME.isMobile() ? 0.35 : 0.4;
+        // let scams = BABYLON.Mesh.CreateCylinder("scam_"+randomPositionChooser, 0.1, scamDiameter, scamDiameter, 16, 0, this.scene);
+        let scams = BABYLON.MeshBuilder.CreateBox("scam_" + randomPositionChooser, {
+            width: scamDiameter,
+            height: scamDiameter,
+            depth: 0.01
+        }, this.scene);
+
+
+        // // Test
+        // var scene = this.scene;
+        // // Dude
+        // BABYLON.SceneLoader.ImportMesh("scam1", "assets/scenes/", "scam.babylon", scene, function (newMeshes2, particleSystems2, skeletons2) {
+        //     var scam = newMeshes2[0];
+        //     scam.position = new BABYLON.Vector3(7.025, 0-9, 0+5.5);
+        //     scam.scaling = new BABYLON.Vector3(0.05, 0.05, 0.05);
+        //     console.log(scam)
+        //     return scene;
+        // }, function (scene) {
+        //     // Called during inprogress
+        //     console.log("InProgress")
+        // }, function (scene, message, exception) {
+        //     // Called when Error Occurs
+        //     console.log("onError");
+        // });
+
+        scams.material = this.level.getMaterial('scamMaterial');
+        scams.position.x = positionX;
+        scams.position.y = 3;
+        scams.position.z = 0;
+
+        if (type == 'ZIG_ZAG') {
+            scams.animations.push(this.createZigZagScamAnimation(scams));
+        } else if (type == 'NORMAL_SCAM') {
+            scams.animations.push(this.createScamAnimation());
+        }
+        else if (type == 'SPEEDY') {
+            scams.animations.push(this.createSpeedyScamAnimation());
+        }
+        else if (type == 'ACCELERATOR') {
+            scams.animations.push(this.createAcceleratorScamAnimation());
+        }
+        else if (type == 'BLACK_OUT') {
+            scams.animations.push(this.createBlackoutAnimation());
+        }
+        else if (type == 'DIAGONAL') {
+            scams.animations.push(this.createDiagonalScamAnimation(scams));
+        }
+        let scamAnimation = this.scene.beginAnimation(scams, 0, 2000, false);
+        var trigger = setInterval(() => {
```

Comment:
is this the best practice method for detecting collisions? It seems surprising to me that you need to manually set a 5ms interval. Looking at the babylon docs, there are a couple of examples where they use `scene.registerBeforeRender` to add code that is triggered before each render cycle. And then they use functions like `if (object1.intersectsMesh(object21, false)) { ...` - see https://doc.babylonjs.com/babylon101/intersect_collisions_-_mesh